### DISCUSSION TITLE
Бигфарма

### DISCUSSION THREAD

**Discussion by TonitaN** [TonitaN/TFL2021#15] [Announcements]
1. Начинаем в 12 часов, обе группы.
2. Собственно фарма длится до 15 часов. После этого небольшой перерыв и решение задач на скорость.

Чуть заранее перед 12 часами я выкладываю в репозиторий список задач. Бронь на задачи и решённые задачи, а также временно отстранённые от брони участники фармы будут отмечаться в общей таблице. Никто ведь не против Google sheets? Если есть возражения, то расшаривать не буду, а буду проектировать на экран с локального файла.  

### Правила Бигфармы

1. Каждый участник имеет право сдать не больше одной задачи стоимостью 1 балл, не больше двух задач стоимостью 2 балла, не больше 3 задач стоимостью 3 балла, не больше четырёх задач стоимостью 4 балла. Всего участник имеет право сдать 5 задач, при этом каждая следующая сданная задача должна быть по стоимости не меньше, чем предыдущая.
2. Задачи поделены на три большие категории: регулярные языки, КС-языки, прочее. Из 5 задач не может быть больше, чем 3 задачи из одной категории. Из 3 подряд сданных задач не могут быть все из одной и той же категории. Хорошо продумайте стратегию действий, если хотите нафармить много баллов.
3. Пока забронированная участником задача не будет им решена, он не вправе бронировать следующую. Но может отказаться от брони - в этом случае он получает штрафной срок 20 минут, в течение которого не может ничего бронировать.
4. За хаотическое поведение участник фармы может получить отстранение от проверки его задачи на 20 минут. За повторное нарушение - отстранение от проверки задачи на час. Хаотическое поведение: не соблюдать очередь на сдачу, злоупотреблять вопросами (будет личный счёт на особо активных любителей задавать вопросы), мешать другим участникам фармы и проверяющему. Красная карточка (полное отстранение от фармы): пиратские операции в гугл таблицах (отменять бронь без уведомления, вмешиваться в чужие брони). Давайте минимизируем кровавую анархию общими усилиями - в противном случае я буду медленнее проверять задачи и пострадают все.

Сразу же предупреждаю по поводу автоматов-ловушек и грамматик-ловушек (без терминальных правил): такие ответы будут засчитываться как 0 баллов. Даже если в остальном всё правильно! Потому что на РК было уж слишком много компроматов на эту тему, в том числе и от коллег, кто должен был помнить наизусть, что так делать не надо 👀 Но если в фарме я вас просто отправлю такой ответ исправлять, то на экзамене и в решении на скорость шансов выбраться из лично сделанной ловушки у вас уже не будет ☠

**Comment by Ankalot** [comment]
1) Самому надо записываться в таблицу для бронирования задачи?
2) Куда отправлять решения задач (если заочное участие)?

**Reply by Ankalot** [reply]
Записываться самому, отправлять на почту.

**Comment by TonitaN** [comment]
Условия открыты. Ссылка на таблицу:
https://docs.google.com/spreadsheets/d/1asUNOFyffXcNIdvjfc9UNly_VG4jdJMFynTy-AWz9Gs/edit?usp=sharing

**Comment by TonitaN** [comment]
Внимание, если вы берёте задачу на много баллов сразу первой, то следующие должны быть по стоимости не меньше её. Вдруг кто-то не обратил внимание.

**Comment by TonitaN** [comment]
Посылаем все решения на почту до 15-10

**Comment by TonitaN** [comment]
Постараюсь проверить всю фарму и всю гонку сегодня, но зависит от количества другой работы. Максимум в понедельник всё будет точно. И ещё +3 балла новогодних сюрпризов хорошим мальчикам и девочкам, про которые сегодня сказала на фарме.
